ext/mbstring/mbstring.c

```diff
@@ -2953,6 +2953,47 @@ PHP_FUNCTION(mb_strtolower)
 	RETURN_STR(mbstring_convert_case(PHP_UNICODE_CASE_LOWER, ZSTR_VAL(str), ZSTR_LEN(str), enc));
 }
 
+static void php_mb_ulcfirst(INTERNAL_FUNCTION_PARAMETERS, php_case_mode mode)
+{
+	zend_string *str, *from_encoding = NULL;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_STR(str)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(from_encoding)
+	ZEND_PARSE_PARAMETERS_END();
+
+	const mbfl_encoding *enc = php_mb_get_encoding(from_encoding, 2);
+	if (!enc) {
+		RETURN_THROWS();
+	}
+
+	zend_string *first = mb_get_substr(str, 0, 1, enc);
+	zend_string *second = mb_get_substr(str, 1, MBFL_SUBSTR_UNTIL_END, enc);
+	zend_string *head = mbstring_convert_case(mode, ZSTR_VAL(first), ZSTR_LEN(first), enc);
+
+	if (zend_string_equals(first, head)) {
+		RETVAL_STR(zend_string_copy(str));
```

Comment:
Note that RETVAL_STR, unlike RETURN_STR, doesn't return; it only sets the return value.
So the code below this if will still execute.

Reply:
Thanks, I mistaken it. fixed.